### PR TITLE
Replace `requests` mocking with `pook`

### LINKED ISSUE #2256: Replace `requests` mocking with `pook`
## Current Situation

`pook` is a library introduced in the API test suite. We have an issue to introduce it to the catalog in https://github.com/WordPress/openverse/issues/1938, however there are other places we should refactor in the API test suites as well.

## Suggested Improvement

Refactor the following test modules to replace `requests` mocking with `pook`.

> *Note*
> Some places may not be able to rely entirely on `pook`, if we're testing weird `requests` edge cases that cannot be triggered by changes in the HTTP response. Those should be explicitly documented when they're needed.

- [x] `test_photon`
- [ ] `test_watermark`
- [ ] `test_waveform`
- [ ] `test_image_views`
- [x] `test_slack` (ingestion server)

## Benefit

Using `pook` makes the expected HTTP response more explicit, meaning the tests are clearer. Mocking `requests` (because we mock the various HTTP verb methods, `get`, `post`, `head`, etc) relies on understanding the specific behaviour requests exhibits under certain HTTP request lifecycle conditions. Using `pook` removes that reliance, creating a test suite that is more resilient long-term to changes in the behaviour of the `requests` library. It also reduces our reliance on `requests` as the specific library we use, which could change if we switch to ASGI Django, for example.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #2256 by @sarayourfriend 

## Description

This PR completes the transition from `requests` mocking to `pook`.

## Testing Instructions

The CI tests (specifically the unit tests) for the API should pass.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/test/unit/utils/test_watermark.py [hunk 1]
```diff
@@ -14,55 +12,33 @@
...
+    pook.on()
+    mock = (
+        pook.get("http://example.com/")
+        .header("User-Agent", HEADERS["User-Agent"])
+        .reply(200)
+        .body(_MOCK_IMAGE_BYTES, binary=True)
+        .mock
     )
+    yield mock
+    pook.off()
```
Comment (sarayourfriend) [inline api/test/unit/utils/test_watermark.py:26]:
```suggestion
def mock_request():
    with pook.use():
        mock = (
            pook.get("http://example.com/")
            .header("User-Agent", HEADERS["User-Agent"])
            .reply(200)
            .body(_MOCK_IMAGE_BYTES, binary=True)
            .mock
        )
        yield mock
```

It's unlikely, but if creating the mock raised an exception then `pook.on` could be left on, never cleaned up.

File: api/test/unit/utils/test_waveform.py [hunk 2]
```diff
@@ -16,42 +12,22 @@
...
+    mock = (
+        pook.get("http://example.org/")
+        .header("User-Agent", UA_STRING)
+        .reply(200)
+        .headers({"Content-Type": _MOCK_AUDIO_INFO["headers"]["Content-Type"]})
+        .body(_MOCK_AUDIO_BYTES, binary=True)
+        .mock
     )
+    yield mock
+    pook.off()
```
Comment (sarayourfriend) [inline api/test/unit/utils/test_waveform.py:27]: Ditto here:

```suggestion
    pook.use():
        mock = (
            pook.get("http://example.org/")
            .header("User-Agent", UA_STRING)
            .reply(200)
            .headers({"Content-Type": _MOCK_AUDIO_INFO["headers"]["Content-Type"]})
            .body(_MOCK_AUDIO_BYTES, binary=True)
            .mock
        )
        yield mock
```

**Review by sarayourfriend — APPROVED** [review]
LGTM! This is great to finally get rid of that hacky requests mocking. Just two non-blocking comments.

**Review by krysal — APPROVED** [review]
Quite nice to see how tests are simplified with this change!
